pkg/operator/ceph/object/zone/controller.go

```diff
@@ -396,3 +450,72 @@ func (r *ReconcileObjectZone) updateStatus(observedGeneration int64, name types.
 	}
 	logger.Debugf("object zone %q status updated to %q", name, status)
 }
+func (r *ReconcileObjectZone) deleteZone(objContext *object.Context) error {
+	realmArg := fmt.Sprintf("--rgw-realm=%s", objContext.Realm)
+	//	zoneGroupArg := fmt.Sprintf("--rgw-zonegroup=%s", objContext.ZoneGroup)
+	zoneArg := fmt.Sprintf("--rgw-zone=%s", objContext.Zone)
+
+	args := []string{"zone", "delete", realmArg, zoneArg}
+	output, err := object.RunAdminCommandNoMultisite(objContext, false, args...)
+	if err != nil {
+		return errors.Wrapf(err, "failed to delete ceph zone %q for reason %q", objContext.Zone, output)
+	}
+	return nil
+}
+
+func (r *ReconcileObjectZone) removeZoneFromZonegroup(objContext *object.Context) error {
+	realmArg := fmt.Sprintf("--rgw-realm=%s", objContext.Realm)
+	zoneGroupArg := fmt.Sprintf("--rgw-zonegroup=%s", objContext.ZoneGroup)
+	zoneArg := fmt.Sprintf("--rgw-zone=%s", objContext.Zone)
+
+	args := []string{"zonegroup", "remove", realmArg, zoneGroupArg, zoneArg}
+	output, err := object.RunAdminCommandNoMultisite(objContext, false, args...)
+	if err != nil {
+		return errors.Wrapf(err, "failed to delete ceph zone %q for reason %q", objContext.Zone, output)
+	}
+
+	args = []string{"period", "update", "--commit", realmArg, zoneGroupArg}
+	output, err = object.RunAdminCommandNoMultisite(objContext, false, args...)
+	if err != nil {
+		return errors.Wrapf(err, "failed to commit updates in ceph zonegroup %q for reason %q", objContext.ZoneGroup, output)
+	}
+	return nil
+}
+
+func (r *ReconcileObjectZone) deletePoolsAndZone(objContext *object.Context, zone *cephv1.CephObjectZone, realmName string) error {
```

Comment:
Separate into `deleteZone()` and `deleteZonePools()`

Reply:
@BlaineEXE  Currently there are two helper functions called from this, for deleting the zone deleteZone() and for deleting pools object.DeletePools(). Am I missing something? Do I need to add another wrapper function around object.DeletePools?

Reply:
Casey's comment here captures my intent: https://github.com/rook/rook/pull/10728#discussion_r961688583